apps/io-services-cms-webapp/src/publicator/request-publication-handler.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  FSMStore,
+  ServiceLifecycle,
+  ServicePublication,
+} from "@io-services-cms/models";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import { Json } from "io-ts-types";
+import { withJsonInput } from "../lib/azure/misc";
+
+const parseIncomingMessage = (
+  queueItem: Json
+): E.Either<Error, ServiceLifecycle.definitions.Service> =>
+  pipe(
+    queueItem,
+    ServiceLifecycle.definitions.Service.decode,
+    E.mapLeft(flow(readableReport, (_) => new Error(_)))
+  );
+
+export class RequestPublicationParseError extends Error {
+  public kind = "RequestPublicationParseError";
+  constructor() {
+    super(`Error while parsing incoming message`);
+  }
+}
```

Comment:
IMHO there is no need to create a custom error, use just  a `new Error("Error while parsing incoming message")`

Reply:
c00ca73